src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessorImplementation.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.ParallelUtils;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Memory;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Transforms
+{
+    /// <summary>
+    /// Provides methods that allow the resizing of images using various algorithms.
+    /// Adapted from <see href="http://www.realtimerendering.com/resources/GraphicsGems/gemsiii/filter_rcg.c"/>
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal class ResizeProcessorImplementation<TPixel> : TransformProcessorBase<TPixel>
```

Comment:
@JimBobSquarePants I wonder whether I should keep the name `ResizeProcessor<TPixel>` instead. For certain processors there will be no pixel-agnostic variant, therefore it might be better to use a uniform naming.

Reply:
I prefer that over Implementation. File name can be ResizeProcessorTPixel.cs I think and pass stylecop naming rules.